### ISSUE TITLE
Godot 4 and Joystick support

### ISSUE THREAD

**Issue by johnnyneverwalked** [drkitt/godot-input-buffer#1]
Hello, I found this addon and it was really useful for my projects!
Since I am working with Godot 4 for on one of them, I forked your repo and made the following changes:

- Converted the buffer to Godot 4
- Added support for Input `InputEventJoypadMotion` events to support gamepad joystick buffering

I could open a PR to merge these changes if you're ok with it, but I didn't want to do it on `main` before asking since it would break 3.x compatibility

**Comment by drkitt** [comment]
I'm glad you found this useful! :D I think both of those changes are helpful but I don't want to break compatibility with 3.x. Would it be possible for you to open a PR that only includes the joystick support, so that this repo can use that feature while keeping compatibility with 3.x? No worries if it's overly difficult to separate the joystick support from 4.0 support. 

Either way, I appreciate the contributions and I'll edit my readme to include a link to your fork.

**Comment by johnnyneverwalked** [comment]
No, that sounds great! I will separate them and open a PR for the joystick support only!
